src/zap.spec.ts

```diff
@@ -0,0 +1,93 @@
+import { ChainId, getZapInDetails } from ".";
+import { EthAddress, ZeroAddress } from "./helpers";
+import { createMockToken, createMockVaultMetadata } from "./test-utils/factories";
+
+describe("Zap", () => {
+  describe("getZapInDetails", () => {
+    it("should return `false` if token does not support zaps", () => {
+      const token = createMockToken({ supported: undefined });
+      const vaultMetadata = createMockVaultMetadata();
+
+      const actual = getZapInDetails({ chainId: 1, token, vaultMetadata });
+
+      expect(actual).toEqual({ isZapInSupported: false });
+    });
+
+    it("should return `false` if vaultMetadata is not given", () => {
+      const token = createMockToken({ supported: { zapperZapIn: true } });
+
+      const actual = getZapInDetails({ chainId: 1, token, vaultMetadata: null });
+
+      expect(actual).toEqual({ isZapInSupported: false });
+    });
+
+    it("should return `false` when the chain is unsupported", () => {
+      const token = createMockToken({ supported: { zapperZapIn: true } });
+      const vaultMetadata = createMockVaultMetadata({ zapInWith: "zapperZapIn" });
+
+      const actual = getZapInDetails({ chainId: 42 as ChainId, token, vaultMetadata });
+
+      expect(actual).toEqual({ isZapInSupported: false });
+    });
+
+    describe("when is Ethereum", () => {
+      it('should return `true` and `zapInWith: "zapperZapIn"` when token is ETH', () => {
```

Comment:
We should remove/change this test, since ETH should be treated as any other token

Reply:
When I run this locally and tried to deposit ETH it was failing because the `vaultMetadata` did not have a `zapInWith`

Reply:
We were missing the `zapInWith` which we get now by calling the whole vault, so we can safely refactor this.

Done with https://github.com/yearn/yearn-sdk/pull/280/commits/00434d7995ea3fb00a4bf3c8eefc1e90c795677d